File: ClipMobile/src/main/java/com/migueljteixeira/clipmobile/ui/ConnectClipActivity.java
package com.migueljteixeira.clipmobile.ui;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;
import android.widget.Toast;

import com.crashlytics.android.Crashlytics;
import com.migueljteixeira.clipmobile.settings.ClipSettings;

import io.fabric.sdk.android.Fabric;

public class ConnectClipActivity extends FragmentActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Fabric.with(this, new Crashlytics());


        // If the user is already logged in, start the StudentNumbersActivity instead
        if( ClipSettings.isUserLoggedIn(this) ) {
            Intent intent = new Intent(getApplicationContext(), StudentNumbersActivity.class);
            startActivity(intent);

            Toast.makeText(this, "you're already logged in, dude!", Toast.LENGTH_SHORT).show();
            finish();
        }

        FragmentManager fm = getSupportFragmentManager();
        ConnectClipFragment fragment = (ConnectClipFragment) fm.findFragmentById(android.R.id.content);

        if (fragment == null) {
            fragment = new ConnectClipFragment();
            fm.beginTransaction().add(android.R.id.content, fragment).commit();
        }
    }

}

File: ClipMobile/src/main/java/com/migueljteixeira/clipmobile/ui/ConnectClipActivity.kt
package com.migueljteixeira.clipmobile.ui

import android.content.Intent
import android.os.Bundle
import com.migueljteixeira.clipmobile.R
import com.migueljteixeira.clipmobile.settings.ClipSettings.getYearSelected
import com.migueljteixeira.clipmobile.settings.ClipSettings.isUserLoggedIn
import com.migueljteixeira.clipmobile.ui.NavDrawerActivity

class ConnectClipActivity : BaseActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_singlepane)
        setupActionBar()

        // Crash system reporting
//        Fabric.with(this, new Crashlytics());

//        Crashlytics.log("ConnectClipActivity - onCreate");

        // If the user has already login, start the StudentNumbersActivity instead
        if (isUserLoggedIn(this)) {
//            Crashlytics.log("ConnectClipActivity - user has already login");
            val intent: Intent = if (getYearSelected(applicationContext) == null) {
                Intent(this, StudentNumbersActivity::class.java)
            } else {
                Intent(this, NavDrawerActivity::class.java)
            }
            startActivity(intent)
            finish()
        }
        val fm = supportFragmentManager
        var fragment = fm.findFragmentById(R.id.content_frame)
        if (fragment == null) {
            fragment = ConnectClipFragment()
            fm.beginTransaction().add(R.id.content_frame, fragment).commit()
        }
    }
}